serve/mlc_serve/engine/staging_engine_worker.py

```diff
@@ -318,6 +324,25 @@ def _adjust_batch(self):
                 self.cache_manager.allocate(state.request_id, num_tokens)
                 self.current_batch[state.request_id] = state
 
+            if not self.current_batch:
+                if len(self.queue) > 0:
+                    LOG.warn(
+                        f"The engine has {len(self.queue)} requests to be processed in the queue, but"
+                        " none of them were added to the current batch during the execution of"
+                        " StagingEngine._adjust_batch"
+                    )
+
+                    hung_request_ids = []
+
+                    for state in self.queue:
+                        hung_request_ids.append(state.request_id)
+                        # TODO(masahi): Proper error enum?
+                        state.validation_err = ValidationError("Canceled due to a hang")
```

Comment:
this error will be returned back to the user, is not better to explain wider that it is internal server error, if we consider it as server hang

Reply:
I don't understand what you are suggesting.

Reply:
I propose to return an error which will explain user what happen. In general there could be several types of error - problem of the request, problem of the model, problem of the server. And getting the error should figure out what happen. `Cancel due to a hang` returned to user most likely will say something to the user, but from my point of view, pointing prefix "Internal Server Error: Request canceled due to hang in the server" will give to user more info

Reply:
ok updated